include/zephyr/drivers/video.h

```diff
@@ -827,6 +827,30 @@ void video_buffer_release(struct video_buffer *buf);
  * @}
  */
 
+/**
+ * @brief Get number of bytes per pixel of a pixel format
+ *
+ * @param pixfmt FourCC pixel format value (\ref video_pixel_formats).
+ */
+static inline unsigned int video_pix_fmt_bpp(uint32_t pixfmt)
+{
+	switch (pixfmt) {
+	case VIDEO_PIX_FMT_BGGR8:
+	case VIDEO_PIX_FMT_GBRG8:
+	case VIDEO_PIX_FMT_GRBG8:
+	case VIDEO_PIX_FMT_RGGB8:
+		return 1;
+	case VIDEO_PIX_FMT_RGB565:
+	case VIDEO_PIX_FMT_YUYV:
+		return 2;
+	case VIDEO_PIX_FMT_XRGB32:
+	case VIDEO_PIX_FMT_XYUV32:
+		return 4;
+	default:
+		return 0;
+	}
+}
```

Comment:
I proposed a `video_bits_per_pixel()` here: https://github.com/zephyrproject-rtos/zephyr/pull/79476
But it is even better if a PR on improving the driver safely moves the utility function into a the shared source.

A `video_byte_per_pixel()` cannot support 10-bit or 12-bit formats? These are well-standardized by i.e. MIPI which defines the bit packing.

If using to mean "bits per pixel", this would work as the multiplication is done before the division:
```c
fmt.pitch = fmt.width * video_bits_per_pixel(fmt.pixelformat) / 8;
```

![scrot_20241023_122633_778x229](https://github.com/user-attachments/assets/b90937b5-68cf-45bb-b85f-716358e91744)

Reply:
Yes, I am aware of your PR and agree it is better. Here, we will use this helper function in several places in this PR to change frame rate so a quick refactor / cleanup is needed unless reviewers will say "code duplication".

Feel free to change this if this PR is merged first or I can rebase our PR if yours is merged first.